runtime/shibuya/src/lib.rs

```diff
@@ -751,8 +751,46 @@ pub type Executive = frame_executive::Executive<
     frame_system::ChainContext<Runtime>,
     Runtime,
     AllPalletsWithSystem,
+    DappsStakingFixEraLength,
 >;
 
+// Migration for fixing era length in dapps staking.
+pub struct DappsStakingFixEraLength;
+
+impl OnRuntimeUpgrade for DappsStakingFixEraLength {
+    fn on_runtime_upgrade() -> frame_support::weights::Weight {
+        use frame_system::pallet::Config;
+        use pallet_dapps_staking::{Config as DappStakingConfig, CurrentEra, NextEraStartingBlock};
+        use sp_runtime::SaturatedConversion;
+
+        // We can always use this formula to calculate the offset to compute the next era starting block
+        // ```rust
+        // let DAPP_STAKING_BLOCK_OFFSET = current_block - (current_block % blocks_per_era) - (current_block * blocks_per_era);
+        // ```
+
+        let dapps_staking_block_offset: u32 = 172800; // Shibuya: 819166 - (819166 % 1200) - (538 * 1200)
```

Comment:
So this is the block at which era 1 started?

Reply:
Yes